CMakeLists.txt

```diff
@@ -0,0 +1,423 @@
+    list(GET ARGN 0 var_default)
+  endif()
+  if(DEFINED ${var_name})
+    set(${var_name} ${${var_name}} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=${${var_name}}")
+  elseif(DEFINED ENV{${var_env}})
+    set(${var_name} $ENV{${var_env}} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=$ENV{${var_env}} (from envvar '${var_env}')")
+  elseif(DEFINED var_default)
+    set(${var_name} ${var_default} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=${var_default} (from default value)")
+  else()
+    message(VERBOSE "legate.core: not setting ${var_name}")
+  endif()
+endfunction()
+
+# Initialize these vars from the CLI, then fallback to an envvar or a default value.
+set_or_default(Legion_SPY USE_SPY OFF)
+set_or_default(Legion_USE_LLVM USE_LLVM OFF)
+set_or_default(Legion_USE_CUDA USE_CUDA OFF)
+set_or_default(Legion_USE_HDF5 USE_HDF OFF)
+set_or_default(Legion_USE_GASNet USE_GASNET OFF)
+set_or_default(Legion_USE_OpenMP USE_OPENMP OFF)
+set_or_default(Legion_BOUNDS_CHECKS CHECK_BOUNDS OFF)
+set_or_default(Legion_MAX_DIM LEGION_MAX_DIM 4)
+
+# Check the max dimensions
+if ((Legion_MAX_DIM LESS 1) OR (Legion_MAX_DIM GREATER 9))
+  message(FATAL_ERROR "The maximum number of Legate dimensions must be between 1 and 9 inclusive")
+endif()
+
+set_or_default(Legion_MAX_FIELDS LEGION_MAX_FIELDS 256)
+
+# Check that max fields is between 32 and 4096 and is a power of 2
+if (NOT Legion_MAX_FIELDS MATCHES "^(32|64|128|256|512|1024|2048|4096)$")
+  message(FATAL_ERROR "The maximum number of Legate fields must be a power of 2 between 32 and 4096 inclusive")
+endif()
+
+set_or_default(GASNet_ROOT_DIR GASNET)
+set_or_default(NCCL_DIR NCCL_PATH)
+set_or_default(Thrust_DIR THRUST_PATH)
+set_or_default(CUDA_TOOLKIT_ROOT_DIR CUDA)
+set_or_default(CMAKE_CUDA_ARCHITECTURES GPU_ARCH ALL)
+
+option(Legion_SPY "Enable detailed logging for Legion Spy" OFF)
+option(Legion_USE_LLVM "Use LLVM JIT operations" OFF)
+option(Legion_USE_CUDA "Enable Legion support for the CUDA runtime" OFF)
+option(Legion_USE_HDF5 "Enable support for HDF5" OFF)
+option(Legion_USE_GASNet "Enable the distributed GASNet backend" OFF)
+option(Legion_USE_OpenMP "Use OpenMP" OFF)
+option(Legion_USE_Python "Use Python" OFF)
+
+if(Legion_USE_GASNet AND (NOT GASNet_ROOT_DIR))
+  option(Legion_EMBED_GASNet "Embed a custom GASNet build into Legion" ON)
+endif()
+
+set_or_default(GASNet_CONDUIT CONDUIT "udp")
+
+option(LEGATE_CORE_EXCLUDE_LEGION_FROM_ALL "Exclude Legion targets from legate.core's 'all' target" OFF)
+
+include(CMakeDependentOption)
+cmake_dependent_option(Legion_HIJACK_CUDART
+  "Allow Legion to hijack and rewrite application calls into the CUDA runtime"
+  ON
+  Legion_USE_CUDA;"DEFINED ENV{USE_CUDART_HIJACK} AND $ENV{USE_CUDART_HIJACK} STREQUAL 1"
+  OFF)
+
+if(Legion_USE_CUDA)
+  if(BUILD_SHARED_LIBS)
+    if(Legion_HIJACK_CUDART)
+      # Must statically link CUDA if HIJACK_CUDART is set
+      set(Legion_CUDA_DYNAMIC_LOAD OFF)
```

Comment:
do you want to add logic for `REALM_USE_CUDART_HIJACK` here as well?

Reply:
Legion [sets](https://github.com/trxcllnt/legion/blob/5da839977d14fc06163db231b80e002a80781bc2/CMakeLists.txt#L364-L366) `REALM_USE_CUDART_HIJACK` if `Legion_HIJACK_CUDART` is set.